shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/data/remote/model/SurveyApiModel.kt

```diff
@@ -0,0 +1,37 @@
+package co.nimblehq.ic.kmm.suv.data.remote.model
+
+import co.nimblehq.ic.kmm.suv.data.local.model.SurveyRealmObject
+import co.nimblehq.ic.kmm.suv.domain.model.Survey
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class SurveyApiModel(
+    val id: String,
+    val type: String,
+    val title: String,
+    val description: String,
+    @SerialName("is_active")
+    val isActive: Boolean,
+    @SerialName("cover_image_url")
+    val coverImageUrl: String
+)
+
+fun SurveyApiModel.toSurvey() = Survey(
+    title,
+    description,
+    isActive,
+    coverImageUrl
+)
+
+
+fun SurveyApiModel.toSurveyRealmObject(): SurveyRealmObject {
+    val survey = SurveyRealmObject()
+    survey.id = id
+    survey.type = type
+    survey.title = title
+    survey.description = description
+    survey.isActive = isActive
+    survey.coverImageUrl = coverImageUrl
+    return survey
```

Comment:
I'd suggest using `apply` in here for `setter(s)` 😄

Reply:
I tried with apply, but because we have the same properties for both `SurveyApiModel` and `SurveyRealmObject`, we can't use this one 🤔

<img width="616" alt="Screenshot 2022-11-21 at 11 42 46" src="https://user-images.githubusercontent.com/19943832/202967354-477a997a-315e-49a9-a966-9aba33f4b899.png">

Reply:
@suho Can we switch to use open class to define Realm model instead? https://github.com/realm/realm-java/blob/master/examples/kotlinExample/src/main/kotlin/io/realm/examples/kotlin/model/Person.kt

```
open class SurveyRealmObject(
    @PrimaryKey
    var id: String = "",
    var type: String = "",
    var title: String = "",
    var description: String = "",
    var isActive: Boolean = false,
    var coverImageUrl: String = ""
) : RealmObject
```

then we can use apply as usual

```suggestion
fun SurveyApiModel.toSurveyRealmObject() = SurveyRealmObject(
    id = id,
    type = type,
    title = title,
    description = description,
    isActive = isActive,
    coverImageUrl = coverImageUrl
)
```

Reply:
in case you can't or don't want to use `open class`, we can even write it like this

```
fun SurveyApiModel.toSurveyRealmObject(): SurveyRealmObject {
    val survey = this
    return SurveyRealmObject().apply {
        id = survey.id
        type = survey.type
        title = survey.title
        description = survey.description
        isActive = survey.isActive
        coverImageUrl = survey.coverImageUrl
    }
}
```

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/102/commits/90445f1173cd31c1ed11874b276208ea228c3957

Reply:
I just run the test, it still occurs this issue `[Realm] Cannot find primary zero arg constructor`. You can see on [the CI](https://github.com/suho/kmm-ic/actions/runs/3520288230/jobs/5901063739)

Reply:
I found this comment: https://github.com/realm/realm-kotlin/issues/184#issuecomment-1037120571

Technically, we need to provide a default zero arg constructor

Reply:
I did with this solution: https://github.com/suho/kmm-ic/pull/102/commits/894c617d6153500da5bf9c8d13d2785f02b5a236

Reply:
@suho That works 👍